### PR TITLE
Add Bleson adapter

### PR DESCRIPTION
**Pull request by ttu**

### CODE REVIEW

**Comment by sergioisidoro** [comment]
I'll post some of the issues I've found. Sorry the spam, I'll post things as I'll find them, and I'll try to fix and contribute with a PR. 

- rx package v3 has major breaking changes. The package in setup.py is not pinned, so this is breaking things: https://rxpy.readthedocs.io/en/latest/migration.html#packages-renamed

in  `ruuvi_rx.py` it should be `from rx.subject import Subject` now (notice that subject is now singular)

File: ruuvitag_sensor/data_formats.py [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+
+def get_raw_bleson(raw, data_format):
+    # Bleson drops FF from the raw data and it is required for DF 3 and 5
+    # TODO: Move convert_data to adaptor specific code
+    if data_format in (2, 4):
+        return raw
+    return 'FF' + raw
+
+
+get_raw = get_raw_bleson if os.environ.get('RUUVI_BLE_ADAPTER') == 'Bleson' else get_raw_hci
```
Comment (sergioisidoro) [inline ruuvitag_sensor/data_formats.py]: Did you consider this approach from one of my old branches:
```
        #Python duck typing
        try:
            string_data = raw.hex()
            if string_data[0:2] != 'FF':
                string_data = 'FF' + string_data

        except (AttributeError):
            string_data = raw
```

Feels a bit wrong do do env checks here.

Comment (ttu) [reply ruuvitag_sensor/data_formats.py]: Yes, I hate using env checks all around the code base. In my opinion env checks should only be done in single place and all functionality should be configured/decided there.

I considered that duck typing, but I don't think that is a good option as when not using Bleson there is lots of unnecessary exceptions. Also using exceptions is not a good way to control the program flow, although in here it is the simplest solution. Better to decide the parsing function to use once.

I changed the functionality a little and use patching. Although don't like to override methods like this, but I think for now this is the best solution. Don't want to make too much refactoring in this commit, so with this I don't have to make changes to multiple places.

**Comment by sergioisidoro** [comment]
One more thing, this one a bit cryptic.
```
Traceback (most recent call last):
  File "examples/find_tags.py", line 5, in <module>
    from ruuvitag_sensor.ruuvi import RuuviTagSensor
  File "/Users/sergio/Development/ruuvitag-sensor/venv/lib/python3.7/site-packages/ruuvitag_sensor-0.13.0-py3.7.egg/ruuvitag_sensor/ruuvi.py", line 15, in <module>
ModuleNotFoundError: No module named 'ruuvitag_sensor.adapters'
```

I've had this problem in some other experiments with home assistant last week, and was chasing my tail for a couple of hours and didn't find what was going on. I wonder if there's some problem with the packaging because I have no problem running the interpreter locally and importing:

```
Python 3.7.6 (default, Dec 30 2019, 19:38:28) 
[Clang 11.0.0 (clang-1100.0.33.16)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from ruuvitag_sensor.adapters import *
>>> 
```
My guess was that it was some old pycache or pyc problem, but didn't find anything wrong. Is it about that 

How to replicate: 
* Setup virtual env with python 3 
```
virtualenv -p python3 venv  
source source venv/bin/activate
```
* Install the package with `python3 setup.py install`
* Run `python3 examples/find_tags.py`

**Comment by ttu** [comment]
@sergioisidoro _ModuleNotFoundError_ was a packaging problem. Added `ruuvitag.adapters` to _setup.py_
https://github.com/ttu/ruuvitag-sensor/blob/add-bleson-adaptor/setup.py#L37

**Pull request merged**